api/app.py

```diff
@@ -14,19 +14,68 @@
 
 mysql = MySQL(app)
 
+@app.route('/api/building/create', methods = ["POST"])
+def create_building():
+    # info is a dictionary with keys address, pet_friendly, laundry_availability, type_of_unit, and distance_from_waterloo
+    # address VARCHAR(255) NOT NULL,
+    # pet_friendly TINYINT(1) NOT NULL,
+    # laundry_availability ENUM('building', 'ensuite', 'none') NOT NULL,
+    # type_of_unit ENUM('apartment','house') NOT NULL,
+    # distance_from_waterloo DECIMAL(3,1) NOT NULL,
 
+    building_info = request.get_json() # {"address": "123 H", "pet_friendly": 0, "laundry_availability": "ensuite", "type_of_unit": "house", "distance_from_waterloo": 0.4}
+    conn = mysql.connection
+    cur = conn.cursor()
 
-@app.route('/api')
-def get_message():
-    return {'message': 'Hello from the API'}
+    try:
+        cur.execute("INSERT INTO Building VALUES (NULL, %s, %s, %s, %s, %s)", 
+                    (building_info["address"], building_info["pet_friendly"], building_info["laundry_availability"], building_info["type_of_unit"], building_info["distance_from_waterloo"]))
+        cur.close()
+        conn.commit()
+        return {"status": True}
+    except Exception as e:
+        return {"status": False, "message": f"Error with inserting: {e}"} 
 
-@app.route('/api/db')
-def get_data():
+    
+@app.route('/api/building/get', methods = ["GET"])
+def get_buildings():
```

Comment:
Would it make sense to add an authentication check here (if the user is logged in)? Or will routes be protected on the FE so they wouldn't be able to access the page.

Reply:
no this is simply for getting all buildings, i didnt create an endpoint for only getting buildings listed by a particular user

Reply:
for this route no since any user can view a building, but for the create unit route we should add a check once the auth pr gets pushed